docs/contributing/vscode-devcontainer.rst

```diff
@@ -0,0 +1,98 @@
+3. Mount your current local directory inside the container
+4. Configure VScode to seamlessly integrate with the container so that Devcontainer enabled extensions use the container directly. For example, opening the terminal will open a bash session directly in the container in your mounted tripal directory.
+5. Install and configure a number really helpful VSCode extensions to make your life easier including php intelephense (php syntax checking and autocomplete), Drupal specific autocomplete, code standards checking, phpunit testing integration, etc. (see more below)
+
+Dev Container: Requirements
+---------------------------
+
+1. Visual Studio Code with the Microsoft Dev Containers extension (`ms-vscode-remote.remote-containers`).
+2. Docker installed locally. We suggest Docker Desktop for Mac and Windows machines.
+
+Dev Container: Usage
+---------------------
+
+1. Clone the Tripal repository locally and open it in VSCode.
+2. You should see a popup in the bottom right corner that lets you know the folder contains a Dev Container configuration file. You should click on "Reopen in Container" which will build a docker image based on the dockerfile in your local Tripal clone, start a container and configure VScode based on the devcontainer.json file. This will take a bit of time.
+
+  .. image:: vscode.devcontainer.notification.png
+
+
+  You will see a "Configuring DevContainer (show log)" notification. The log will show you the docker build progress and upon completion prints out the site admin login information.
+
+  When it's complete, you will see the DevContainer status in the bottom left corner which indicates you are using "Dev Container @ TripalDocker".
+
+3. If you don't see the popup above, you can instead click on the Dev Container status icon in the bottom left corner. This will open the command palette where you can choose "Reopen in Container".
+
+  .. image:: vscode.devcontainer.statusIcon.png
+
+  .. image:: vscode.devcontainer.commandpalette.png
+
+Tour of Devcontainer Features/Benefits
+---------------------------------------
+
+Terminal mounts inside container
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In VSCode, go to Terminal in the top menu bar and click "New Terminal". This will open a terminal inside the container which allows you to use Drush natively without the need for "docker exec" or any other command prefixes!
+
+.. image:: vscode.devcontainer.terminal.drush.png
+
+Ports automatically mapped
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Ports exposed via the dockerfile are now mapped automatically to ports on your local machine. This allows you to have multiple devcontainers open without having to think about ports at all.
+
+To view your Drupal site, go to "Ports", hoveer over the forwarded address for port 80 which runs the webserver and then click on the little globe icon. This will open the Drupal homepage in your default web browser. Note the login information was printed to the terminal during devcontainer startup *winks*.
```

Comment:
```suggestion
To view your Drupal site, go to "Ports", hover over the forwarded address for port 80 which runs the webserver, and then click on the little globe icon. This will open the Drupal homepage in your default web browser. Note the login information was printed to the terminal during devcontainer startup 😉.
```
Did you mean the winks emoji here?

Reply:
Yup but I didn't know how to add it in and didn't want to take the time to look it up.